[css-counter-styles] cjk-earthly-branch and cjk-heavenly-stem should have 'fixed' system (258089)

vitorroriz opened the pull request:
#### 096e15ff5659dfcd7c87134955b9371e04f385b5
<pre>
[css-counter-styles] cjk-earthly-branch and cjk-heavenly-stem should have &apos;fixed&apos; system (258089)
<a href="https://bugs.webkit.org/show_bug.cgi?id=258089">https://bugs.webkit.org/show_bug.cgi?id=258089</a>
rdar://110796633

Reviewed by Tim Nguyen.

According to spec, cjk-earthly-branch and cjk-heavenly-stem counter-styles
should have &apos;fixed&apos; system instead of &apos;alphabetic&apos;.
See w3c/csswg-drafts#8596

* LayoutTests/imported/w3c/web-platform-tests/css/css-counter-styles/cjk-earthly-branch/css3-counter-styles-202-expected.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-counter-styles/cjk-earthly-branch/css3-counter-styles-202.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-counter-styles/cjk-heavenly-stem/css3-counter-styles-205-expected.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-counter-styles/cjk-heavenly-stem/css3-counter-styles-205.html:
* Source/WebCore/css/counterStyles.css:
(@counter-style cjk-earthly-branch):
(@counter-style cjk-heavenly-stem):

Canonical link: <a href="https://commits.webkit.org/265235@main">https://commits.webkit.org/265235@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/4f3bd423bdeaac184295f63c9587123881bb2695

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/10338 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/10572 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/10843 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/11982 "Built successfully") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/32/builds/9932 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/10349 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/12597 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/10524 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/12897 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/10495 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/15/builds/11253 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/8710 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/12377 "Built successfully") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/39/builds/8556 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/9364 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/16630 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/9653 "Passed tests") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/36/builds/9515 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/12759 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/9954 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/17/builds/8082 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/9113 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/9259 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/2478 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/13364 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/9805 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

vitorroriz commented on LayoutTests/imported/w3c/web-platform-tests/css/css-counter-styles/cjk-earthly-branch/css3-counter-styles-202-expected.html:
```diff
@@ -5,38 +5,39 @@
 <title>cjk-earthly-branch, 13+</title>
 <link rel='author' title='Richard Ishida' href='mailto:ishida@w3.org'>
 <link rel='help' href='http://www.w3.org/TR/css-counter-styles-3/#simple-numeric'>
+<link rel='match' href='css3-counter-styles-202-ref.html'>
```
ops, I'm removing this.

vitorroriz commented on LayoutTests/imported/w3c/web-platform-tests/css/css-counter-styles/cjk-heavenly-stem/css3-counter-styles-205-expected.html:
```diff
@@ -5,39 +5,40 @@
 <title>cjk-heavenly-stem, 10+</title>
 <link rel='author' title='Richard Ishida' href='mailto:ishida@w3.org'>
 <link rel='help' href='http://www.w3.org/TR/css-counter-styles-3/#simple-numeric'>
+<link rel='match' href='css3-counter-styles-205-ref.html'>
```
ditto.

brentfulgham commented on Source/WebCore/css/counterStyles.css:
```diff
@@ -286,16 +286,15 @@
     /* ▸ */
 }
 
-/* cjk-earthly-branch and cjk-heavenly-stem are using 'alphabetic' system instead of 'fixed', as in the spec, for compatibility with legacy behavior and WPT tests (https://github.com/w3c/csswg-drafts/issues/8596). */
```
Could you add a comment with the Issue link where you and the WG folks decided this was the right change?

brentfulgham reviewed (commented):
Looks good to me, but could you have @nt1m double-check the WPT bits?

nt1m reviewed (approved):
2 comments:
- It would be nice to test numbers that are covered by the "fixed" range
- Can you backport your fixes from the WPT PR (the comments you addressed there) to the WebKit PR?

vitorroriz committed:
[css-counter-styles] cjk-earthly-branch and cjk-heavenly-stem should have 'fixed' system (258089)
https://bugs.webkit.org/show_bug.cgi?id=258089
rdar://110796633

Reviewed by Tim Nguyen.

According to spec, cjk-earthly-branch and cjk-heavenly-stem counter-styles
should have 'fixed' system instead of 'alphabetic'.
See w3c/csswg-drafts#8596

* LayoutTests/imported/w3c/web-platform-tests/css/css-counter-styles/cjk-earthly-branch/css3-counter-styles-202-expected.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-counter-styles/cjk-earthly-branch/css3-counter-styles-202.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-counter-styles/cjk-heavenly-stem/css3-counter-styles-205-expected.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-counter-styles/cjk-heavenly-stem/css3-counter-styles-205.html:
* Source/WebCore/css/counterStyles.css:
(@counter-style cjk-earthly-branch):
(@counter-style cjk-heavenly-stem):

Canonical link: https://commits.webkit.org/265235@main
